### ISSUE TITLE
Impossible to acquire new token if token in localStorage is expired

### ISSUE THREAD

**Issue by foxylion** [AzureAD/microsoft-authentication-library-for-js#3078]
## Library
- [x] `@azure/msal-browser@2.11.2`

## Framework

- [x] Other

## Error Message

The request to `https://login.microsoftonline.com/common/oauth2/v2.0/token` results in the following error:

```
{
  "error": "invalid_grant",
  "error_description": "AADSTS70000: The user could not be authenticated as the grant is expired. The user must sign in again.\r\nTrace ID: da470717-ec3a-413b-b595-f351f70e0200\r\nCorrelation ID: f0dc8d84-1664-4e51-987e-91fd2ff21083\r\nTimestamp: 2021-02-24 19:23:11Z",
  "error_codes": [70000],
  "timestamp": "2021-02-24 19:23:11Z",
  "trace_id": "da470717-ec3a-413b-b595-f351f70e0200",
  "correlation_id": "f0dc8d84-1664-4e51-987e-91fd2ff21083",
  "error_uri": "https://login.microsoftonline.com/error?code=70000",
  "suberror": "bad_token"
}
```

## MSAL Configuration

```js
  private msalConfig: Configuration = {
    auth: {
      clientId: '...',
      redirectUri: 'https://localhost:3000,
    },
    cache: {
      cacheLocation: 'localStorage',
    },
  };
```

## Reproduction steps

1. Sign-In through `this.application.acquireTokenRedirect({ scopes: this.scopes });`
2. Handle response with `const result = await this.application.handleRedirectPromise();`
3. Use the access token with
  ```
const result = await this.application.acquireTokenSilent({
        account: this.application.getAllAccounts()[0],
        scopes: this.scopes,
      });
```

Up until here everything works fine.

1. Wait for 24 hours
2. `application.acquireTokenSilent` will now return NULL (expected)
3. Try to renew session with `app.acquireTokenRedirect` and `app.handleRedirectPromise` (same code as before)
4. calling handleRedirectPromise will now return NULL (not expected)
5. The browser request log contains the above mentioned failed token request

## Expected behavior

Even when the current token in local storage is expired (what I expect) the combination of `acquireTokenRedirect` and `handleRedirectPromise` should result in a new token returned and not end in an error.

Clearing the browsers local storage fix the problem until the next hours of inactivity passed.

## Identity Provider

- [x] Microsoft Consumer Account

## Browsers/Environment

- [x] Chrome (88.0.4324.182) on Ubuntu

## Regression

- I don't know, this is the first time I try using this library

## Security

- no

## Source

- [x] Customer request

**Comment by pkanher617** [comment]
> application.acquireTokenSilent will now return NULL (expected)

This is not expected - acquireTokenSilent should be failing with and error when it can't retrieve a new set of tokens (due to token expiration). Could you add a try/catch or `.catch` to the acquireTokenSilent call and check whether an error is surfacing?

Could you post a snippet of how you are calling `acquireTokenRedirect` and how you are handling the redirect promise?

**Comment by forellendev** [comment]
You only need to add the account field to your redirect request:

`
const result = await this.application.acquireTokenSilent({
      account: this.application.getAllAccounts()[0]
});
`

**Comment by foxylion** [comment]
@forellendev I tried to add `account: this.application.getAllAccounts()[0]` to my `acquireTokenRedirect` call, that did not help.

But it looks like I found out the issue that was preventing `handleRedirectPromise` from succeeding.

In another part of the code I called `acquireTokenSilent` before the code had a possibility to run `handleRedirectPromise` this seemed to break the code behind `handleRedirectPromise`.

So I think it is solved. If not I will come back. :-)
